src/sema/expression/literals.rs

```diff
@@ -518,6 +518,53 @@ pub(super) fn struct_literal(
     }
 }
 
+pub(crate) fn unit_literal(
+    loc: &pt::Loc,
+    unit: &Option<Identifier>,
+    ns: &mut Namespace,
+    diagnostics: &mut Diagnostics,
+) -> BigInt {
+    if let Some(unit) = unit {
+        match unit.name.as_str() {
+            "wei" | "gwei" | "ether" if ns.target != crate::Target::EVM => {
+                diagnostics.push(Diagnostic::warning(
+                    *loc,
+                    "ethereum currency unit used while not targetting ethereum".to_owned(),
+                ));
+            }
+            "sol" | "lamports" if ns.target != crate::Target::Solana => {
+                diagnostics.push(Diagnostic::warning(
+                    *loc,
+                    "Solana currency unit used while not targetting Solana".to_owned(),
+                ));
+            }
+            _ => (),
+        }
+
+        match unit.name.as_str() {
+            "seconds" => BigInt::from(1),
+            "minutes" => BigInt::from(60),
+            "hours" => BigInt::from(60 * 60),
+            "days" => BigInt::from(60 * 60 * 24),
+            "weeks" => BigInt::from(60 * 60 * 24 * 7),
+            "wei" => BigInt::from(1),
+            "gwei" => BigInt::from(10).pow(9u32),
+            "ether" => BigInt::from(10).pow(18u32),
+            "sol" => BigInt::from(10).pow(9u32),
+            "lamports" => BigInt::from(1),
```

Comment:
@xermicus what should we do for substrate here, if anything?

Reply:
The currency type can be configured in the contracts pallet, so I don't think we should do anything for now.

Reply:
I was worried that was going to be the answer.